src/window.h

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#ifndef WINDOW_H
+#define WINDOW_H
+
+#include "graphics.h"
+
+struct window {
+    int x;
+    int y;
+    int width;
+    int height;
+    struct graphics_color border_color;
+    struct graphics_color background_color;
+};
+
+/**
+ * @brief Creates a window for given dimens
+ * @details Creates a window with the given location and size
+ *
+ * @param x The x position of the window, relative to the screen
+ * @param y The y position of the window, relative to the screen
+ * @param width The width of the window in pixels
+ * @param height The height of the window in pixels
+ * @return A pointer to the window
+ */
+struct window *window_create(int x, int y, int width, int height);
+
+/**
+ * @brief Draws a line in the window
+ * @details Draws a line at given position in the window
+ * It will be clipped if it extends outside the window
+ * All positions are realtive to the window
+ * @param window The window to draw in
+ * @param x1 The x position of the first point in the line
+ * @param y1 The position of the first point in the line
+ * @param x2 The x position of the second point in the line
+ * @param y2 The y position of the second point in the line
+ * @param color The color of the line
+ */
+void window_draw_line(struct window *w, int x1, int y1, int x2, int y2, struct graphics_color color);
+
+/**
+ * @brief Draws an arc in the window
+ * @details Draws an arc with given size and center inside the window
+ * Curves existing outside the window bounds will be clipped. All points
+ * relative to the window
+ *
+ * @param window The window to draw in
+ * @param x The x position of the center of the arc
+ * @param y The y position of the center of the arc
+ * @param r The radius of the arc
+ * @param start_theta The starting angle of the arc
```

Comment:
These angles must be between 0 and 2PI or things might get weird, as they are passed into graphics_arc which passes them directly into our math.h's cos() function which has this requirement on input angles. We can add that restriction to the comments in both graphics.h and window.h, or we can add that robustness to the math library.